source/CardGames.py

```diff
@@ -101,6 +117,19 @@ def addCard(self, card: Card, isKnown: bool = True):
     else:
       self.knownCards.append(False)
 
+  def setHand(self, cards: "list[Card]", isKnown: bool = False):
+    self.hand = cards
+    self.knownCards = [isKnown for _ in self.hand]
+
+  #function
+  def MatchFour(self):
+    for combination in itertools.combinations(self.hand, 4):
+      if all(card.value == combination[0].value for card in combination):
+        print("You have a set of %ss!" % combination[0].value)
+        self.addSetOfFour(combination[0].value)
+        return True
+    return False
+
```

Comment:
Maybe leave comments explaining what the itertools is and what your function is doing so that it is more clear to understand. It's not too long and gets the job done, so good job.

Reply:
Ok! I will add in comments for this by next sprint and will implement this from now on!